### PR TITLE
HOSTEDCP-807: Check KAS loadbalancer health

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
This enhances HCAvailable condition to better represent day2 state

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-807](https://issues.redhat.com/browse/HOSTEDCP-807)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -660,6 +666,57 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
...
+			return fmt.Errorf("failed to get kube apiserver route: %w", err)
+		}
+
+		if len(internalRoute.Status.Ingress) == 0 || internalRoute.Status.Ingress[0].RouterCanonicalHostname == "" {
+			return fmt.Errorf("APIServer load balancer is not provisioned")
+		}
+		hostName = internalRoute.Status.Ingress[0].RouterCanonicalHostname
+	} else {
+		var svc *corev1.Service
+		if serviceStrategy.Type == hyperv1.LoadBalancer && util.IsPrivateHCP(hcp) {
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:689]: If the hcp is publicAndPrivate I think we'd want to report on both load balancers' health

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 2]
```diff
@@ -660,6 +666,57 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
...
 
+func (r *HostedControlPlaneReconciler) healthCheckKASExternalEndpoint(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
+	serviceStrategy := util.ServicePublishingStrategyByTypeForHCP(hcp, hyperv1.APIServer)
+	if serviceStrategy == nil {
+		return fmt.Errorf("APIServer service strategy not specified")
+	}
+
+	var hostName string
+	if serviceStrategy.Type == hyperv1.Route {
+		// svc = manifests.RouterPublicService(hcp.Namespace)
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:677]: did you mean to remove this?

**Review by csrwng — COMMENTED** [review]
Some comments. Thx!

File: test/e2e/util/oauth.go [hunk 3]
```diff
@@ -106,7 +106,7 @@ func WaitForOAuthToken(t *testing.T, ctx context.Context, oauthRoute *routev1.Ro
 	}
 
 	var access_token string
-	err = wait.PollImmediateWithContext(ctx, time.Second, time.Minute, func(ctx context.Context) (done bool, err error) {
+	err = wait.PollImmediateWithContext(ctx, time.Second, time.Minute*2, func(ctx context.Context) (done bool, err error) {
```
Comment (enxebre) [inline test/e2e/util/oauth.go:109]: why this change?

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 4]
```diff
@@ -665,6 +670,68 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	return result, nil
 }
 
+func (r *HostedControlPlaneReconciler) healthCheckKASLoadBalancers(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:673]: nit:  a comment documenting this func would be nice.

File: test/e2e/util/oauth.go (hunk 3 above)
Comment (muraee) [reply test/e2e/util/oauth.go:109]: oauth idp test was failing, Cesar suggested that the configuration are not propagated in time before this test times out.

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 5]
```diff
@@ -665,6 +670,70 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	return result, nil
 }
 
+// healthCheckKASLoadBalancers pefroms a health check on the KubeAPI server /healthz endpoint using the the public and private load balancers hostnames directly
```
Comment (bryan-cox) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:673]: nit: spelling typo
```suggestion
// healthCheckKASLoadBalancers performs a health check on the KubeAPI server /healthz endpoint using the the public and private load balancers hostnames directly
```
